internal/service.go

```diff
@@ -50,56 +55,54 @@ func NewHauserService(config *config.Config, fsClient client.DataExportClient, s
 		fsClient: fsClient,
 		storage:  storage,
 		database: db,
-	}
-}
-
-func headerName(columnName string) string {
-	if bundleField, ok := bundleFieldsMap[columnName]; !ok {
-		return columnName
-	} else {
-		return bundleField.Name
+		schema:   warehouse.MakeSchema(warehouse.BaseExportFields{}),
 	}
 }
 
 // TransformExportJSONRecord transforms the record map (extracted from the API response json) to a
 // slice of strings. The slice of strings contains values in the same order as the existing export table.
 // For existing export table fields that do not exist in the json record, an empty string is populated.
-func TransformExportJSONRecord(convert warehouse.ValueToStringFn, tableColumns []string, rec map[string]interface{}) ([]string, error) {
+func (h *HauserService) transformExportJSONRecord(convert warehouse.ValueToStringFn, rec map[string]interface{}) ([]string, error) {
 	var line []string
 	lowerRec := make(map[string]interface{})
 	customVarsMap := make(map[string]interface{})
 
+	if len(h.schemaMap) == 0 {
+		h.schemaMap = make(map[string]bool, len(h.schema))
```

Comment:
I did it here because the schema can change between `NewHauser()` and actually working on the first download due to needing to reconcile with an existing database. Once the process has ben started though, the schema doesn't change. 

Instead of doing it in both the initializer and the when initializing the database, I did it here. I could see a case for pulling this out into something like `func (h *Hauser) setSchema(s Schema)` that rebuilds this map.

Reply:
Eh, I guess it's fine. Maybe we just need a comment then?